numpy/core/src/multiarray/number.c

```diff
@@ -814,213 +814,112 @@ _array_nonzero(PyArrayObject *mp)
     }
 }
 
-
+/*
+ * Convert the array to a scalar if allowed, and apply the builtin function
+ * to it. The where argument is passed onto Py_EnterRecursiveCall, in case
+ * there is
+ */
 NPY_NO_EXPORT PyObject *
-array_int(PyArrayObject *v)
+array_scalar_forward(PyArrayObject *v,
+                     PyObject *(*builtin_func)(PyObject *),
+                     const char *where)
```

Comment:
In principle we could use templating instead of function pointers, but we don't really care about speed here.